client/src/components/Ticketing/ticketingmanager/AdminPurchase/ReaderPurchase.tsx

```diff
@@ -0,0 +1,234 @@
+      window.removeEventListener('beforeunload', alertUser);
+    };
+  }, []);
+
+  const handleCloseDialog = () => {
+    setDialog(false);
+    navigate('/ticketing/purchaseticket');
+  };
+
+  const handleRefresh = async () => {
+    try {
+      const stripe = await stripePromise;
+      if (!stripe) throw new Error('Failed to initialize stripe!');
+
+      const {paymentIntent} = await stripe.retrievePaymentIntent(clientSecret);
+      if (!paymentIntent) throw new Error('Cannot find payment intent!');
+
+      const newStatus = 'payment_intent.' + paymentIntent.status;
+      setStatus(newStatus);
+
+      if (newStatus === 'payment_intent.succeeded') {
+        const ws = getWebSocket();
+        ws.close();
+        navigate(`/success`);
+      }
+    } catch (error) {
+      console.error(error.message);
+      setErrMsg(error.message);
+      setDialog(true);
+    }
+  };
+
+  const handleCancel = async () => {
+    let newStatus = '';
+    try {
+      const stripe = await stripePromise;
+      if (!stripe) throw new Error('Failed to initialize stripe!');
+
+      const {paymentIntent} = await stripe.retrievePaymentIntent(clientSecret);
+      if (!paymentIntent) throw new Error('Cannot find payment intent!');
+
+      newStatus = 'payment_intent.' + paymentIntent.status;
+      setStatus(newStatus);
+    } catch (error) {
+      console.error(error.message);
+      setErrMsg(error.message);
+      setDialog(true);
+    }
+    if (newStatus != 'terminal.reader.action_succeeded' && // must make sure we haven't already finished charging
+        newStatus != 'payment_intent.succeeded' &&
+        newStatus != 'charge.succeeded') {
+      if (!token) return;
+      const response = await fetch( // request payment and put order in database
+        process.env.REACT_APP_API_2_URL + `/order/reader-cancel`,
+        {
+          credentials: 'include',
+          method: 'POST',
+          headers: {
+              'Content-Type': 'application/json',
+              'Authorization': `Bearer ${token}`,
+            },
+          body: JSON.stringify({paymentIntentID}),
+        },
+      );
+
+      if (!response.ok) {
+        throw response;
+      }
+
+      console.log('order cancelled!');
+      navigate('/ticketing/purchaseticket');
+    } else {
+      setErrMsg('Order completed before cancelation could occur.');
+      setDialog(true);
+    }
+  };
+
+  return (
+    <>
+      <div className='w-full h-screen overflow-x-hidden absolute'>
+        <div className='md:ml-[18rem] md:mt-40 md:mb-[11rem] tab:mx-[5rem] mx-[1.5rem] my-[9rem]'>
+          <h1 className='font-bold text-5xl bg-clip-text text-transparent bg-gradient-to-r from-green-500 to-zinc-500 mb-14'>
+            Reader Order Status
+          </h1>
+          <div className='text-3xl font-semibold'>
+            {'ID: ' + paymentIntentID}
+          </div>
+          <div className='text-3xl font-semibold'>
+            {'Status: ' + status}
+          </div>
```

Comment:
I think these make sense for debugging purposes, but for practical use, I think we wouldn't need the `paymentIntentID` displayed. And then regarding the status, we could make this a bit more human-friendly by mapping the possible values of status to less code-looking values. Ex. `payment_intent.requires_payment_method` --> `Requires Payment Method`

Reply:
I decided to replace the payment Intent ID here with the order number, since its less visually busy and still contains potentially useful information, statuses are also now "translated" to normal looking text

Reply:
Sounds good, thanks!